server/src/test/java/org/opensearch/index/query/QueryStringQueryBuilderTests.java

```diff
@@ -427,6 +428,12 @@ public void testTimezoneEquals() {
         assertNotEquals(builder1, builder2);
     }
 
+    public void testExtractAllUsedFields() {
```

Comment:
The tests are insufficient, please cover different query types

Reply:
on it!

Reply:
pushed tests